iceoryx2-bb/posix/src/signal.rs

```diff
@@ -75,32 +75,54 @@ use tiny_fn::tiny_fn;
 
 macro_rules! define_signals {
     {fetchable: $($entry:ident = $nn:ident::$value:ident),*
+     fatal_fetchable: $($fatal_entry:ident = $fatal_nn:ident::$fatal_value:ident),*
      unknown_translates_to: $unknown_entry:ident = $unknown_nn:ident::$unknown_value:ident
      unfetchable: $($uentry:ident = $unn:ident::$uvalue:ident),*
     } => {
         /// Represents signals which cannot be fetched by a signal handler.
-        #[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
+        #[derive(Debug, Clone, Copy, Eq, Hash, PartialEq, Sequence)]
         #[repr(i32)]
         pub enum UnfetchableSignal {
           $($uentry = $unn::$uvalue),*,
         }
 
+        enum_gen! {
+            /// Represents signals that can be fetched and are non-fatal, meaning that they do not
+            /// indicate that the process is somehow corrupted. For example: `SIGUSR1` or `SIGINT`.
+            #[derive(Sequence)]
+            NonFatalFetchableSignal
+          unknown_translates_to:
+            $unknown_entry = $unknown_nn::$unknown_value
+          entry:
+            $($entry = $nn::$value),*
+        }
+
+        /// Represents signals that can be fetched and are fatal, meaning that they
+        /// indicate that the process is somehow corrupted and should terminate.
+        #[derive(Debug, Clone, Copy, Eq, Hash, PartialEq, Sequence)]
+        #[repr(i32)]
+        pub enum FatalFetchableSignal {
```

Comment:
Nothing to change, but just out of curiosity - why the distinction of being `Fetchable` ? What is a `NonFetchableSignal` ?

Reply:
For `NonFetchableSignal` you cannot register a custom callback via `sigaction`. For instance `SIGKILL` does not send anything to the process but the OS directly kills the process without informing it.